Port Gaeguli to GStreamer 1.16

xhaakon opened the pull request.

xhaakon committed:
pipeline: replace FIFO with srtsink

xhaakon committed:
meson: require GStreamer >= 1.16.0

xhaakon committed:
pipeline: allow passing 'username' parameter

xhaakon committed:
tests: disable test-fifo-transmit

Some cases will be moved to pipeline tests and the rest dropped.

xhaakon commented:
@justinjoy This is merely FYI now. Still needs some work in the terms of the test suite.

justinjoy commented:
> @justinjoy This is merely FYI now. Still needs some work in the terms of the test suite.

No problem. Just tell me when you think these are ready. :)

Just a quick note. As you can see, CI node at Azure was dropped. 
All of repository is transiting to github action so some of unit tests could be failed due to wrong CI environment settings.

xhaakon committed:
gaeguli: add gaeguli_pipeline_target_get_bytes_sent()

xhaakon committed:
meson: don't split stdout and stderr in debug logs

It splits GStreamer log from the Gaeguli one, which makes it hard to see
connections between both.

xhaakon committed:
tests: add pipeline-add-remove-target-random

Replaces fifo-transmit-add-remove-random

xhaakon committed:
pipeline: handle errors when srtsink can't bind to local address

xhaakon committed:
tests: add pipeline-address-in-use

xhaakon committed:
tests: use create_receiver in pipeline tests

Copied from fifo-transmit tests, will be reused by new pipeline cases.

xhaakon committed:
tests: add pipeline-listener

xhaakon committed:
tests: add pipeline-listener-random

xhaakon committed:
Remove fifo-transmit

xhaakon committed:
Update README

justinjoy commented on README.md:
```diff
@@ -1,14 +1,11 @@
-*   fifo-transmitter: Reads fifo and send streaming to a specific URI using SRT protocol
-
-So basically generation and transmission are decoupled.
+*Gaeguli* implements the supporting library for handling SRT streaming. It provides 'pipeline' component
+that deals with receiving frames from a video source and streaming to a specific URI using SRT protocol.
```
👍
